client/client-lib/src/lib.rs

```diff
@@ -453,7 +453,17 @@ impl<T: AsRef<ClientConfig> + Clone> Client<T> {
         self.reissue(all_coins, rng).await
     }
 
-    pub async fn await_consensus_block_height(&self, block_height: u64) -> u64 {
+    pub async fn await_consensus_block_height(
+        &self,
+        block_height: u64,
+    ) -> std::result::Result<u64, task::Elapsed> {
+        task::timeout(Duration::from_secs(30), async {
```

Comment:
At minimum - we should have timeouts for any network operations, at least on the client side. It's better to return an error and have user retry, than just hang forever with no feedback to the user.